From accdf729747b97bb95a899d5541c137fbf0c2c14 Mon Sep 17 00:00:00 2001
From: Ramneek Mehresh <ramneek.mehresh@freescale.com>
Date: Fri, 5 Dec 2014 17:56:02 +0530
Subject: [PATCH 094/124] drivers:usb:gadget: Fix kernel boot dump for usb
 gadegt drv

Copy of_node property only for host mode so as to fix gadget mode
kernel dump messages

Signed-off-by: Ramneek Mehresh <ramneek.mehresh@freescale.com>
Change-Id: Ia80243d9531629f748c066259da979ba6e59adf1
Reviewed-on: http://git.am.freescale.net:8181/25431
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Suresh Gupta <suresh.gupta@freescale.com>
Reviewed-by: Matthew Weigel <Matthew.Weigel@freescale.com>
---
 drivers/usb/host/fsl-mph-dr-of.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/host/fsl-mph-dr-of.c b/drivers/usb/host/fsl-mph-dr-of.c
index a424a41..79d03e4 100644
--- a/drivers/usb/host/fsl-mph-dr-of.c
+++ b/drivers/usb/host/fsl-mph-dr-of.c
@@ -103,7 +103,8 @@ static struct platform_device *fsl_usb2_device_register(
 	else
 		dma_set_mask(&pdev->dev, DMA_BIT_MASK(32));
 
-	pdev->dev.of_node = ofdev->dev.of_node;
+	if (pdata->operating_mode != FSL_USB2_DR_DEVICE)
+		pdev->dev.of_node = ofdev->dev.of_node;
 
 	retval = platform_device_add_data(pdev, pdata, sizeof(*pdata));
 	if (retval)
-- 
2.1.0.27.g96db324

